pkg/target/target.go

```diff
@@ -426,6 +427,95 @@ func convertToLabelSelector(object map[string]interface{}) (*metav1.LabelSelecto
 	return obj, nil
 }
 
+// TODO: can we use generic for Unmarshal after go 1.18?
+func convertToMatch(object map[string]interface{}) (*match.Match, error) {
+	j, err := json.Marshal(object)
+	if err != nil {
+		return nil, errors.Wrap(err, "Could not convert unknown object to JSON")
+	}
+	obj := &match.Match{}
+	if err := json.Unmarshal(j, obj); err != nil {
+		return nil, errors.Wrap(err, "Could not convert JSON to Match")
+	}
+	return obj, nil
+}
+
+// ToMatcher converts .spec.match in mutators to Matcher.
 func (h *K8sValidationTarget) ToMatcher(u *unstructured.Unstructured) (constraints.Matcher, error) {
-	panic("not implemented")
+	obj, found, err := unstructured.NestedMap(u.Object, "spec", "match")
+	if err != nil {
+		return nil, fmt.Errorf("%w: %v", ErrCreatingMather, err)
+	}
+	if found && obj != nil {
+		match, err := convertToMatch(obj)
+		if err != nil {
+			return nil, fmt.Errorf("%w: %v", ErrCreatingMather, err)
+		}
+		return &Matcher{match}, nil
+	}
+	return nil, fmt.Errorf("%w: %s %s has no match found", ErrCreatingMather, u.GetKind(), u.GetName())
+}
+
+// Matcher implements constraint.Matcher.
+// TODO: add cache when handler.Cache interface is available.
+type Matcher struct {
+	match *match.Match
+}
+
+func (m *Matcher) Match(review interface{}) (bool, error) {
+	switch req := review.(type) {
+	case *gkReview:
+		obj, oldObj, ns, err := gkReviewToObject(req)
+		if err != nil {
+			return false, err
+		}
+		return matchAny(m, ns, &obj, &oldObj)
+	case gkReview:
+		obj, oldObj, ns, err := gkReviewToObject(&req)
+		if err != nil {
+			return false, err
+		}
+		return matchAny(m, ns, &obj, &oldObj)
+	default:
+		return false, fmt.Errorf("%w: expect gkReview, got %T", ErrReviewFormat, review)
+	}
+}
+
+func matchAny(m *Matcher, ns *corev1.Namespace, objs ...*unstructured.Unstructured) (matched bool, err error) {
```

Comment:
There's no need to use named return values here - just use (bool, error) since as-is can have weird side effects. If you want to document what the returned values mean - it's better for that to be in the method comment (I think it's fine with or without a comment)

Reply:
done